code-name-taran/src/com/spaghettisoft/component/chess/gamelogic/GameLogic.java

```diff
@@ -0,0 +1,906 @@
+		int currentColumn = cordinatesFigureAttackKing[ONE];
+		Move move;
+		AbstractFigure figure2;
+		while (true) {
+			
+			for (int i = 0; i < this.chessBoard.getBoardLenght(); i++) {
+				for (int j = 0; j < this.chessBoard.getBoardLenght(); j++) {
+					figure2 = this.chessBoard.getFigure(i, j);
+					if (figure2 != null && !isEnemyFigure(player, i, j) && !isThisFigureKing(figure2)) {
+						move = new Move(i, j, currentRow, currentColumn);
+						if (isThisFigureKnight(i, j)) {
+							if (figure2.validMove(move) && !isKignInCheckWhenMoveFigure(player, move)) {
+								return true;
+							}
+						} else {
+							if (figure2.validMove(move) && !checkForFiguresBetweenTwoFields(move) && !isKignInCheckWhenMoveFigure(player, move)) {
+								return true;
+							}
+						}
+					}
+				}
+			}
+
+			currentRow += direction[ZERO];
+			currentColumn += direction[ONE];
+			if ( !IsRowAndColumAreMatrixRange(currentRow, currentColumn) || (currentRow == kingPosition[ZERO] && currentColumn == kingPosition[ONE]) ) {
+				break;	
+			}
+			
+		}
+		return false;
+	}
+	
+	private boolean isAllFreeFieldsAroundKingAreInCheck(Player player , int[] kingPosition) {
+		int row;
+		int column;
+		AbstractFigure figure;
+		int numberOfFreeFields = 0;
+		int nummberOfFreeFieldsInCheck = 0;
+		
+		for (int i = 0; i < directions.length; i++) {
+			row = kingPosition[ZERO] + directions[i][ZERO];
+			column = kingPosition[ONE] + directions[i][ONE];
+			if (IsRowAndColumAreMatrixRange(row, column)) {
+				figure = this.chessBoard.getFigure(row, column);
+				if (figure == null || isEnemyFigure(player, row, column)) {
+					numberOfFreeFields++;
+					if (isFieldInCheck(player, row, column)) {
+						nummberOfFreeFieldsInCheck++;
+					}
+				}
+			}
+		}
+		
+		return numberOfFreeFields == nummberOfFreeFieldsInCheck;
+		
+	}
+
+	public int getNumberOfFicgureOnTheChessBoard() {
+		return numberOfFicgureOnChessBoard;
+	}
+
+	public void decrementNumberOfFicgureOnTheChessBoard() {
+		 numberOfFicgureOnChessBoard--;
+	}
+	
+	public int getNumberOfMovesWithoutMovePawn() {
+		return numberOfMovesWithoutMovePawn;
+	}
+
+	public void incrementNumberOfMovesWithoutMovePawn() {
+		numberOfMovesWithoutMovePawn++;
+	}
+
+	public void resetNumberOfMovesWithoutMovePawn() {
+		numberOfMovesWithoutMovePawn = 0;
+	}
+
+	public int getNumberOfMovesWithoutCapture() {
+		return numberOfMovesWithoutCapture;
+	}
+
+	public void incrementNumberOfMovesWithoutCapture() {
+		numberOfMovesWithoutCapture++;
+	}
+
+	public void resetNumberOfMovesWithoutCapture() {
+		numberOfMovesWithoutCapture = 0;
+	}
+
+
+	public boolean isCurentPlayerInCheck() {
+		return isCurentPlayerInCheck;
+	}
+
+
+	public void setCurentPlayerInCheck(boolean isCurentPlayerInCheck) {
+		this.isCurentPlayerInCheck = isCurentPlayerInCheck;
+	}
+
+
+	public Move getLastEnemyMove() {
+		return lastEnemyMove;
+	}
+
+
+	public void setLastEnemyMove(Move lastEnemyMove) {
+		this.lastEnemyMove = lastEnemyMove;
+	}
+
+}

```

Comment:
The logic of the chess game is very complex, still your code reads well - Good job :+1: 
If you want to improve the readability of the code further, I suggest you to split the logic in several classes. For example - one class can contain the rules for the king, another for the pawns and so on.